clients/da-vinci-client/src/main/java/com/linkedin/davinci/utils/ChunkAssembler.java

```diff
@@ -0,0 +1,150 @@
+package com.linkedin.davinci.utils;
+
+import com.linkedin.davinci.storage.chunking.AbstractAvroChunkingAdapter;
+import com.linkedin.davinci.storage.chunking.RawBytesChunkingAdapter;
+import com.linkedin.davinci.store.memory.InMemoryStorageEngine;
+import com.linkedin.davinci.store.record.ValueRecord;
+import com.linkedin.venice.compression.NoopCompressor;
+import com.linkedin.venice.compression.VeniceCompressor;
+import com.linkedin.venice.pubsub.api.PubSubTopicPartition;
+import com.linkedin.venice.serialization.RawBytesStoreDeserializerCache;
+import com.linkedin.venice.serialization.StoreDeserializerCache;
+import com.linkedin.venice.serialization.avro.AvroProtocolDefinition;
+import com.linkedin.venice.serializer.RecordDeserializer;
+import com.linkedin.venice.utils.lazy.Lazy;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/*
+ * This class serves as a utility to deserialize and assemble chunks consumed from a Kafka topic 
+ */
+public class ChunkAssembler {
+  private static final Logger LOGGER = LogManager.getLogger(ChunkAssembler.class);
+
+  protected static final VeniceCompressor NO_OP_COMPRESSOR = new NoopCompressor();
+  protected final HashMap<Integer, VeniceCompressor> compressorMap = new HashMap<>();
```

Comment:
Same with this.

Reply:
Thank you for catching these!

Reply:
Should the LOGGER be kept since it's being used in `bufferAndAssembleRecord`?

Reply:
logger is ok to keep I think